extensions/wikia/Search/classes/Services/ESFandomSearchService.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Wikia\Search\Services;
+
+class ESFandomSearchService extends AbstractSearchService {
+	const RESULTS_COUNT = 6;
+
+	protected function prepareQuery( $phrase ) {
+		return urlencode( $phrase );
+	}
+
+	protected function select( $query ) {
+		global $wgConsulServiceTag, $wgConsulUrl;
+
+		$consulUrl =
+			( new \Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl,
+				$wgConsulServiceTag ) )->getUrl( 'fandom-search' );
+
+		$response = \Http::post( "http://$consulUrl/fandom", array(
+			'noProxy' => true,
+			'postData' => "queryTerms=$query",
+			'headers' => [
+				'Content-Type' => 'application/x-www-form-urlencoded',
+			],
+		) );
+		if ( $response !== false ) {
+			$decodedResponse = json_decode( $response, true );
+			if ( json_last_error() !== JSON_ERROR_NONE ) {
+				\WikiaLogger::instance()->error( " Fandom Stories Search: error decoding response", [
+					'query' => $query,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "query" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)